_posts/2022-02-01-android-ui-testing-in-allegro-pa-organization.md

```diff
@@ -0,0 +1,178 @@
+       }.overpaymentItemPage(position = 1) { item ->
+           item.tapOnOverpayment()
+       }
+
+   inOverpayment(parseOverpaymentData(OVERPAYMENT_WITHOUT_IBAN_POSITIVE))
+       .toolbarPage { toolbar ->
+           toolbar.checkName(OVERPAYMENT_TOOLBAR_NAME_RES_ID)
+               .tapToolbarBack()
+       }
+
+   inSettings()
+       .toolbarPage { toolbar ->
+           toolbar.checkName(SETTINGS_TOOLBAR_NAME_RES_ID)
+               .tapToolbarBack()
+       }
+
+   inDashboard()
+       .toolbarPage { toolbar ->
+           toolbar.checkName(DASHBOARD_TOOLBAR_NAME_RES_ID)
+       }
+}
+```
+
+The above test allows you to check the path followed by the user who wants to see the overpayment screen
+from the dashboard and then return. Methods with the `in` prefix contain a page object that provides the
+necessary interactions and assertions to check the view. By running this test on the test device, you can
+see the start of the application with the dashboard screen, where the title of the toolbar is checked
+and the button that takes you to the options screen is clicked. In the options screen, the title of
+the toolbar is checked and you click on the option that takes you to the overpayment screen.
+The toolbar title also is checked in the overpayment screen and the back arrow is clicked. The
+application returns to the options screen, the title of the toolbar is checked and the back arrow
+is clicked. The last screen that is checked is the dashboard screen. And then the test
+passes. The class fragment presented below has the method that was used in the previous test. This method allows
+for creates settings page object and run check on toolbar page and overpayment item page. Toolbar page object
+allows for check name by string resource defined in `strings.xml` file. Whereas `overpaymentItemPage()` method
+searches for item on the first position on the recycler view. After that it clicks on overpayment option on screen.
+This means that the methods can be used for other tests without unnecessarily duplicating it.
+
+```kotlin
+inSettings()
+   .toolbarPage { toolbar ->
+       toolbar.checkName(SETTINGS_TOOLBAR_NAME_RES_ID)
+   }.overpaymentItemPage(position = 1) { item ->
+       item.tapOnOverpayment()
+   }
+```
+
+Our test team proposed to provide test scenarios that should be implemented in addition to the so-called
+happy paths. These are or used to be the most frequently reported problems by our customers regarding the display
+of data or switching between screens. However, it should also be noticed that the number of these tests
+shouldn’t be too large, because running one test usually takes a few seconds, what may cause queues in the
+test environment. Additionally, the tests need to be merged into the main branch. The 10 sample tests took
+almost 30 seconds to complete, but the entire procedure took almost a minute.
+![10 simple tests](/img/articles/2022-02-01-android-ui-testing-in-allegro-pa-organization/10_tests.png)
```

Comment:
```suggestion
![10 sample tests](/img/articles/2022-02-01-android-ui-testing-in-allegro-pa-organization/10_tests.png)
```
Nie wiem, czym simple to adekwatne określenie.

```suggestion
![Test time summary](/img/articles/2022-02-01-android-ui-testing-in-allegro-pa-organization/10_tests.png)
```

Reply:
Sample ;)